Formula/obfs4proxy.rb

```diff
@@ -0,0 +1,18 @@
+class Obfs4proxy < Formula
+  desc "Pluggable transport proxy for Tor, implementing obfs4"
+  homepage "https://gitlab.com/yawning/obfs4"
+  url "https://gitlab.com/yawning/obfs4/-/archive/obfs4proxy-0.0.11/obfs4-obfs4proxy-0.0.11.tar.gz"
+  sha256 "46f621f1d94d244e7b1d0b93dafea7abadb2428f8b1d0463559426362ea98eae"
+  license "BSD-2-Clause"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", "-o", "obfs4proxy/obfs4proxy", "./obfs4proxy"
+    bin.install "obfs4proxy/obfs4proxy" => "obfs4proxy"
+  end
+
+  test do
+    system "#{bin}/obfs4proxy", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Hi thanks for the advice; but as I mentioned – I don’t think a standalone test is possible for this particular program. I’ll try to come up with something, but in the case that this isn’t possible; do let me know if Homebrew won’t accept the PR.

Reply:
> If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
Same if the software requires a virtual machine, docker instance, etc. to be running.

This would also mean if it requires Tor. But indeed, without a reasonable test it won't be merged.

Reply:
Alright I have something figured out. I’ll push the commit by tomorrow.